extensions/2.0/Khronos/KHR_displaymapping_pq/README.md

```diff
@@ -0,0 +1,400 @@
+
+#### Color conversion matrix
+
+Color conversion from BT.709 to BT.2020 is specified in ITU BT.2087  
+https://www.itu.int/rec/R-REC-BT.2087/en
+
+The M2 linear color conversion matrix is defined as:  
+0.6274 0.3293 0.0433  
+0.0691 0.9195 0.0114  
+0.0164 0.0880 0.8956  
+
+A HDR capable display is defined as having at least 10 bits per pixel for each colorchannel. 
+
+### To SDR capable display
+
+If the framebuffer format or colorspace is not known, or none is available that preserves range, precision and color gamut then lower range framebuffer with sRGB colorspace may be used.  
+This is to allow for compatibility with displays that does not support higher range and/or compatible colorspaces.  
+It also allows for implementations where the details of the framebuffer is not known or available.  
+
+A SDR capable display is defined as having less than 10 bits per pixel for each colorchannel.  
+
+For SDR output,  a range extension value of 46.42 and gamma of 2.4 shall be used.  
+
+
+### OOTF
+
+Opto-optical-transfer-function  
+Input signal to the OOTF is the scene linear light. The OOTF maps relative scene linear light to display linear light.  
+For more information see the ITU BT.2100 specification.  
+
+Resulting linear scene light values (the pixel output) shall be display mapped according to the parameter `Reference PQ OOTF` of ITU BT.2100   
+The opto-optical reference transform shall be applied, the reference transform compatible with both SDR and HDR displays is described in ITU BT.2390 with the use of range extension and gamma values.  
+
+E′ = G709[E] = pow(1.099 (rangeExtension * E), 0.45) – 0.099 for 1 > E > 0.0003024
+               267.84 * E for 0.0003024 ≥ E ≥ 0
+FD = G1886[E'] = pow(100 E′, gamma)
+
+E is the linear scene light value. 
+
+Where the rangeExtension and gamma values shall be set according to HDR or SDR display.  
+
+```
+HDR
+range extension = 59.5208  
+gamma = 2.4  
+```
+
+```
+SDR
+range extension = 46.42  
+gamma = 2.4  
+```
+
+
+**Implementation Notes**
+
+Pseudocode for BT.2100 reference OOTF  
+
+```
+BT_2100_OOTF(color, rangeExponent, gamma) {  
+    if (color <= 0.0003024) {  
+        nonlinear = 267.84 * color;  
+    else {  
+        nonlinear = 1.099 * pow(rangeExponent * color, 0.45) - 0.099;  
+    }  
+    return 100 * pow(nonlinear, gamma);
+}  
+```
+
+Where this is calculated per RGB component, note that the color value in this equation must be in range 0.0 - 1.0
+It may be estimated that the display will not have the ability to output dark levels in the region of 0.0003024 in which case implementations may ignore that condition and use the same operator for the whole range 0.0 - 1.0  
+
+The incoming color value shall first be adjusted by aperture as detailed in [See Scene aperture](#scene-aperture) 
+
+
+### OETF
+
+Opto-electrical-transfer-function  
+Input signal to the PQ OETF is the scene linear light. The PQ OETF maps scene linear light to non-linear PQ signal value.  
```

Comment:
Note that according to the spec, the OOTF expects values between 0 and 1 (which means values already exposed and maybe tone mapped by the camera).

Reply:
Correct.
Perhaps I should make it clear that at this stage all values are expected to be in the 0 - 10000 range, as declared by this extension?

Then, depending on how you choose to implement the OOTF and the OETF, one solution is to simply divide by 1000 before applying the OOTF.

Reply:
I have clarified the wording and added implementation notes

Reply:
I also have issues with the requirement of bounding the scene values to the 0-10,000 range :) And dividing by 10,000 wouldn't work well for all scenes.

Reply:
> I also have issues with the requirement of bounding the scene values to the 0-10,000 range :) And dividing by 10,000 wouldn't work well for all scenes.

Could you please elaborate as to why a known dynamic range of 0 - 10000 would be a problem?

Why would it not work well for some scenes?
(Remember, this factor has nothing to do with 'auto exposure', it would be an implementers way of mapping the known dynamic range into the 0 - 1 range. Thats all)

Reply:
I as well think this is quite problematic.  I think that light intensities can be focused or change dynamically (grazing fresnel or via glass refractions) based on camera and scene configurations which may not be known ahead of time.   Thus a hard limit is super hard to enforce.  I am also not sure it is that needed.

Reply:
> (Remember, this factor has nothing to do with 'auto exposure', it would be an implementers way of mapping the known dynamic range into the 0 - 1 range. Thats all)

That's what tone mappers are used for. Since they are usually non-linear, they let you compress the range while preserving contrast intents, which you can't achieve with a simple scalar factor.

Reply:
I Ben and thanks for your comment

> I as well think this is quite problematic - trying to enforce a maximum intensity limit.

It does not enforce a maximum intensity limit, you should not interpret this value as maximum intensity.
It defines the known dynamic range to 0 - 10000, simply because this is the range where the perceptual quantizer will produce gradients without visible banding (assuming display is >= 10 bits per pixel)

For any case where it is desired to have a different (intensity) range the `sceneAperture` can be used to specify that.

Hope this clarifies any confusion.

Reply:
> That's what tone mappers are used for. Since they are usually non-linear, they let you compress the range while preserving contrast intents, which you can't achieve with a simple scalar factor.

I think that you are confusing the input to the reference rendering intent (reference OOTF)  with the non-linear PQ - this takes linear scene values and 'compress' into a non-linear range intended for the output device.

Please read the sections "Perceptual Quantizer - reference OOTF" and "Perceptual Quantizer - reference OETF"